packages/fiber/tests/core/renderer.test.tsx

```diff
@@ -672,4 +682,38 @@ describe('renderer', () => {
     })
     expect((THREE as any).ColorManagement.legacyMode).toBe(false)
   })
+
+  it('can handle createPortal', async () => {
+    const scene = new THREE.Scene()
+
+    let state: RootState = null!
+    let portalState: InjectState = null!
+
+    const Component = () => {
+      const three = useThree()
+      portalState = three
+
+      return <group />
+    }
+
+    await act(async () => {
+      state = createRoot(canvas)
+        .render(createPortal(<Component />, scene, { scene }))
+        .getState()
+    })
+
+    // Renders into portal target
+    expect(scene.children.length).not.toBe(0)
+
+    // Creates an isolated state enclave
+    expect(state.scene).not.toBe(scene)
+    expect(portalState.scene).toBe(scene)
+
+    // Preserves internal keys
+    const overwrittenKeys = ['events', 'internal']
+    const respectedKeys = privateKeys.filter(
+      (key) => overwrittenKeys.includes(key) || state[key] === (portalState as RootState)[key],
+    )
+    expect(respectedKeys).toStrictEqual(privateKeys)
+  })
```

Comment:
@drcmda, I'm using strict equality to compare states here, but is `internal` supposed to be copied over? I see that you use my trick of injecting into the store via the events callback so `events` for sure is overwritten here.

Reply:
Was from a race condition, the returned state from render was newer than that from any rendered component.